src/coffee/stockimport.coffee

```diff
@@ -383,7 +383,8 @@ class StockImport
                 debug "Error on handling 409 stock update error. Details: #{JSON.stringify(err)}"
                 Promise.reject err
           else
-            Promise.reject new Error("Failed to retry the task after #{@max409Retries} attempts for stock #{JSON.stringify(entry)}")
+            debug "Failed to retry the task after #{@max409Retries} attempts for stock #{JSON.stringify(entry)}"
+            Promise.reject err
```

Comment:
Hello @Babanila and welcome :) 

Thank you for addressing the issue. It is important for the library user to know if rejected request was retried or did not qualify for retry at all. This information is important for both log based statistics/alerts as in general for issue debugging/resolution.

As described in the ticket #73 

> I think it'd be better if we extend the error from SDK with ours and rethrow it.

please make sure that a message which currently is output as debug only is also included into rejected error. I am not sure what would be the best practice here as simple re-throw will unfortunately lose original stack trace but it might be good enough if we just extend original error with our custom/context specific message:

```
customMessage = "Failed to retry the task after #{@max409Retries} attempts for stock #{JSON.stringify(entry)}"
err.sphereStockImportErrorMessage = customMessage
Promise.reject err
```

Reply:
Here we try to do the same but i guess we lose original stack trace and possibly other properties from the `e`: https://github.com/commercetools/nodejs/blob/master/packages/sdk-middleware-http/src/http.js#L177